benchmark-service/benchmark/benchmark.py

```diff
@@ -28,21 +73,51 @@ async def greet_user(request: web.Request) -> web.Response:
         'username': request.match_info.get('username', ''),
         'current_date': 'July 10, 2020'
     }
-    response = aiohttp_jinja2.render_template('user.html', request,
+    response = aiohttp_jinja2.render_template('name.html', request,
                                               context=context)
     return response
 
 
+@router.get('/name/{name}')
+@web_authenticated_developers_only(redirect=False)
+async def show_name(request: web.Request, userdata) -> web.Response:  # pylint: disable=unused-argument
+    benchmarks = get_benchmarks(default_filepath)
+    name_data = next((item for item in benchmarks['data'] if item['name'] == str(request.match_info['name'])),
+                     None)
+    try:
+        fig = px.scatter(x=list(range(0, len(name_data['times']))), y=name_data['times'])
+        plot = json.dumps(fig, cls=plotly.utils.PlotlyJSONEncoder)
+    except Exception:
+        message = 'could not find name'
+        log.info('name is of type NoneType: ' + message, exc_info=True)
+        raise web.HTTPBadRequest(text=message)
+
+    context = {
+        'name': request.match_info.get('name', ''),
+        'plot': plot
+    }
+
+    return await render_template('benchmark', request, userdata, 'name.html', context)
+
+
 @router.get('/')
 @router.get('')
 @web_authenticated_developers_only(redirect=False)
 async def index(request: web.Request, userdata) -> Dict[str, Any]:  # pylint: disable=unused-argument
-    context = {
-        'current_date': 'July 10, 2020'
-    }
-    response = aiohttp_jinja2.render_template('index.html', request,
-                                              context=context)
-    return response
+    benchmarks_context = get_benchmarks(default_filepath)
+    return await render_template('benchmark', request, userdata, 'index.html', benchmarks_context)
+
+
+@router.post('/lookup')
+@check_csrf_token
+@web_authenticated_developers_only(redirect=False)
+async def lookup(request, userdata):  # pylint: disable=unused-argument
+    data = await request.post()
+    file = data['file']
+    # global default_filepath
```

Comment:
delete. I think what's causing confusion here is that we don't want our application to have a saved state. What happens if multiple people are looking at benchmarks at the same time? Instead, the information needed to render the response should come from the request in the form of POST fields or GET query parameters. The reason for the default file is just to have an initial starting point for the first time a user loads the page if they don't give a file path in the index page. But for this lookup request, we'll always get the file from the POST request.